src/browser/ui/dom/DOMPropertyOperations.js

```diff
@@ -105,6 +106,11 @@ var DOMPropertyOperations = {
         return '';
       }
       return name + '=' + quoteAttributeValueForBrowser(value);
+    } else if (tagName != null && tagName.indexOf('-') >= 0) {
+      if (value == null) {
+        return '';
+      }
+      return name + '=' + quoteAttributeValueForBrowser(value);
```

Comment:
Yeah – even if that's not likely to be idiomatic code, it's surprising if an odd attribute name results in a security hole. You can see that we already check the tag name when generating markup.